### PR TITLE
update peanut data loader class

### PR DESCRIPTION
**Pull request by jihyunbak**
Replace old loader with my locally updated code.

The three new modules (`data_manager.py`, `binning.py` and `task.py`) form their own package that does not depend on the old `peanut.py`.

I think the functions loosely overlap with Ankit's codes that are currently in `peanut.py`, but not exactly the same. @akumar01 what would be a reasonable way to move forward?

### CODE REVIEW

File: neuropacks/peanut/peanut.py [hunk 1]
```diff
@@ -140,7 +143,7 @@ def load_peanut(fpath, epoch, spike_threshold, bin_width=100, bin_type='time', b
     return dat
 
 
-class Peanut_SingleEpoch():
```
Comment (jihyunbak) [inline neuropacks/peanut/peanut.py:143]: @akumar01 Also have you been using this class at all, or is it safe to replace like this?

**Comment by akumar01** [comment]
Hi @jihyunbak , I would advocate for having a common binning/boxcoxing/filtering code to convert spike times to spike rates that goes in a utils file outside of the peanut folder, as it would be used for other datasets. Perhaps in peanut.py we could stick to the class structure that Jesse described at lunch, and use an abstracted "get_spike_rates" type function that lives somewhere else (I can do this). If this is ok, I think the other functionality you have (e.g. in data_manager.py) that manages experimental metadata in no way conflicts with functionality I am interested in having. I would also like to transfer over some code that segments the epochs into distinct well transitions (not sure if you have code to do this as well), as well as bins spikes by location rather than time (useful for finding place fields). Let me know what you think.

**Comment by jihyunbak** [comment]
Thanks @akumar01 ! That sounds good, In fact I was making some more cleanups on this branch, which I just pushed:
- moving the binning-related code to the common utils
- dropping the old `Peanut_SingleEpoch` class (you are not using this, if I understood correctly?)

Let me know if this disrupts anything.

At this point, the resulting `peanut.py` is in the same state as it was a few months ago -- ideally, the functions in this module can be integrated with the `neuropacks.utils.binning` module. Do you want to take a pass on it?

**Comment by jihyunbak** [comment]
> I would also like to transfer over some code that segments the epochs into distinct well transitions (not sure if you have code to do this as well), as well as bins spikes by location rather than time (useful for finding place fields).

I have code for choosing particular trajectory types based on well transitions; I think the current DataManager can do it, but I'll take a look.
EDIT: I use the `bin_and_parse_traj` method to do this.

I have not tried location based binning, but that's a good idea.

**Comment by akumar01** [comment]
@jihyunbak I will take a pass at the utils.binning code as well as peanut.py in the next few days and get back to you.

**Comment by jihyunbak** [comment]
@akumar01 Great, let me know! Meanwhile I'll go ahead and merge this PR for now, because it only makes changes that work for me & does not impact your existing code. Let's clean up in a separate PR.

**Pull request merged**